src/contracts/core/DelegationManager.sol

```diff
@@ -857,69 +888,35 @@ contract DelegationManager is
      * @notice Returns the number of actively-delegatable shares a staker has across all strategies.
      * @dev Returns two empty arrays in the case that the Staker has no actively-delegateable shares.
      */
-    function getDelegatableShares(address staker) public view returns (IStrategy[] memory, uint256[] memory) {
-        // Get currently active shares and strategies for `staker`
-        int256 podShares = eigenPodManager.podOwnerShares(staker);
-        (IStrategy[] memory strategyManagerStrats, uint256[] memory strategyManagerShares) =
-            strategyManager.getDeposits(staker);
-
-        // Has no shares in EigenPodManager, but potentially some in StrategyManager
-        if (podShares <= 0) {
-            return (strategyManagerStrats, strategyManagerShares);
+    function getDelegatableShares(
+        address staker
+    ) public view returns (IStrategy[] memory, uint256[] memory) {
+        // Get a list of all the strategies to check
+        IStrategy[] memory strategies = strategyManager.getStakerStrategyList(staker);
+        // resize and add beaconChainETH to the end
+        assembly {
+            mstore(strategies, add(mload(strategies), 1))
```

Comment:
This is extremely unsafe assembly. If you do this you need to update the free memory pointer; otherwise it's possible the final element in this list gets overwritten by a future malloc.

Reply:
Shouldn't we be using .pop() and .push() here?